server/partition.go

```diff
@@ -732,15 +732,16 @@ func (p *partition) messageProcessingLoop(recvChan <-chan *nats.Msg, stop <-chan
 // processPendingMessage sends an ack if the message's AckPolicy is LEADER and
 // adds the pending message to the commit queue. Messages are removed from the
 // queue and committed when the entire ISR has replicated them.
-func (p *partition) processPendingMessage(offset int64, msg *commitlog.Message) {
+func (p *partition) processPendingMessage(offset, timestamp int64, msg *commitlog.Message) {
 	ack := &client.Ack{
-		Stream:           p.Stream,
-		PartitionSubject: p.Subject,
-		MsgSubject:       string(msg.Headers["subject"]),
-		Offset:           offset,
-		AckInbox:         msg.AckInbox,
-		CorrelationId:    msg.CorrelationID,
-		AckPolicy:        msg.AckPolicy,
+		Stream:             p.Stream,
+		PartitionSubject:   p.Subject,
+		MsgSubject:         string(msg.Headers["subject"]),
+		Offset:             offset,
+		AckInbox:           msg.AckInbox,
+		CorrelationId:      msg.CorrelationID,
+		AckPolicy:          msg.AckPolicy,
+		ReceptionTimestamp: timestamp,
```

Comment:
Is there a reason you're passing back the timestamps from `commitlog.Append` and then passing them into `processPendingMessage`? I think you could leave the commitlog API unchanged and just do the following here:

```go
ReceptionTimestamp: msg.Timestamp,
```

The `commitlog.Message` has the [receive timestamp on it](https://github.com/liftbridge-io/liftbridge/blob/master/server/commitlog/message.go#L22).

Reply:
Good point. I've updated the PR.